pkg/core/auth/authz/policy/accesskey_test.go

```diff
@@ -4,72 +4,61 @@ import (
 	"net/http"
 	"testing"
 
-	"github.com/skygeario/skygear-server/pkg/core/model"
+	"github.com/skygeario/skygear-server/pkg/core/auth"
+	"github.com/skygeario/skygear-server/pkg/core/config"
 	. "github.com/smartystreets/goconvey/convey"
 )
 
 func TestDenyNoAccessKey(t *testing.T) {
 	Convey("Test DenyNoAccessKey", t, func() {
-		Convey("should return error if auth context has no access key", func() {
-			req, _ := http.NewRequest("POST", "/", nil)
-			ctx := MemoryContextGetter{
-				mAccessKey: model.AccessKey{Type: model.NoAccessKeyType},
-			}
+		req, _ := http.NewRequest("POST", "/", nil)
+		ctx := MemoryContextGetter{}
 
+		Convey("should return error if auth context has no access key", func() {
 			err := DenyNoAccessKey(req, ctx)
 			So(err, ShouldNotBeEmpty)
 		})
 
 		Convey("should not return error if auth context has api key ", func() {
-			req, _ := http.NewRequest("POST", "/", nil)
-			ctx := MemoryContextGetter{
-				mAccessKey: model.AccessKey{Type: model.APIAccessKeyType},
-			}
-
+			req = req.WithContext(auth.WithAccessKey(req.Context(), auth.AccessKey{
+				Client: config.OAuthClientConfiguration{},
+			}))
 			err := DenyNoAccessKey(req, ctx)
 			So(err, ShouldBeEmpty)
 		})
 
-		Convey("should not return error if auth context has master key ", func() {
-			req, _ := http.NewRequest("POST", "/", nil)
-			ctx := MemoryContextGetter{
-				mAccessKey: model.AccessKey{Type: model.MasterAccessKeyType},
-			}
-
+		Convey("should return error if auth context has master key ", func() {
+			req = req.WithContext(auth.WithAccessKey(req.Context(), auth.AccessKey{
+				IsMasterKey: true,
+			}))
 			err := DenyNoAccessKey(req, ctx)
-			So(err, ShouldBeEmpty)
+			So(err, ShouldNotBeEmpty)
```

Comment:
So DenyNoAccessKey does not accept master key?

Reply:
Yes. Most use of access key is to obtain a client configuration (e.g. session configuration in authentication). Master key does not provide these information so it cannot be used.

Reply:
What about changing the name from DenyNoAccessKey to RequireClient?

Reply:
Updated.